source/postprocess/visualization/heating.cc

```diff
@@ -120,17 +125,31 @@ namespace aspect
 
             for (unsigned int c=0; c<this->n_compositional_fields(); ++c)
               in.composition[q][c] = uh[q][this->introspection().component_indices.compositional_fields[c]];
+
+            mid_point += evaluation_points[q]/n_quadrature_points;
           }
 
         in.position = evaluation_points;
 
-        this->get_material_model().evaluate(in, out);
+        typename DoFHandler<dim>::active_cell_iterator cell;
+        cell = (GridTools::find_active_cell_around_point<> (this->get_mapping(), this->get_dof_handler(), mid_point)).first;
```

Comment:
Interesting issue. Presumably other future extensions that add other variables to the system will eventually need this as well. We might want to also extend the `MaterialModelInputs` structure in the same way as the `MaterialModelOutputs` structure. But this can wait till a later time.